integration-tests/run-integration-tests.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+set -e
+
+parallel_runs=50
+
+fail_flag_file="/tmp/test_fail_flag"
+rm -f "$fail_flag_file"
+tmp_dir="temp_logs"
+
+cleanup() {
+    if [ -d "$tmp_dir" ]; then
+        for tmp_log in "$tmp_dir"/*; do
+            if grep -q "FAIL" "$tmp_log"; then
+                echo "Failure detected in $tmp_log:"
+                cat "$tmp_log"
+            fi
+            rm -f "$tmp_log"
+        done
+        rmdir "$tmp_dir"
+    fi
+    docker-compose down
+}
+
+success() {
+    echo "All integration tests passed!"
+    exit 0
+}
+
+make -C .. agent-flow
+AGENT_BINARY_PATH="../../../build/grafana-agent-flow"
+
+docker-compose up -d
+
+mkdir -p "$tmp_dir"
+
+counter=0
+
+# Run tests in parallel
+while read -r test_dir; do
+    (
+        pushd "$test_dir"
+        dir_name=$(basename "$test_dir")
+        agent_logfile="../../${tmp_dir}/${dir_name}_agent.log"
+        test_logfile="../../${tmp_dir}/${dir_name}_test.log"
+        "$AGENT_BINARY_PATH" run config.river > "$agent_logfile" 2>&1 &
```

Comment:
Don't all the agents attempt to open on the same port 12345? How do you avoid this problem?

Reply:
I guess that the first one succeed and the other ones don't but that does not seem to create any problem. I will need to dig into that